addons/website_mass_mailing/controllers/main.py

```diff
@@ -9,42 +9,47 @@
 class MassMailController(main.MassMailController):
 
     @route('/website_mass_mailing/is_subscriber', type='json', website=True, auth="public")
-    def is_subscriber(self, list_id, **post):
-        email = None
-        if not request.env.user._is_public():
-            email = request.env.user.email
-        elif request.session.get('mass_mailing_email'):
-            email = request.session['mass_mailing_email']
+    def is_subscriber(self, list_id, input_name, **post):
```

Comment:
Instead of input_name, would probably support subscription_type or something like that that would be currently "email" and "sms", that way we accept only some valid types and compute field / values / ... based on that.